client/app/bundles/course/learning-map/components/Gate/index.jsx

```diff
@@ -0,0 +1,182 @@
+import { connect } from 'react-redux';
+import { selectGate } from 'course/learning-map/actions';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import { red, green } from '@mui/material/colors';
+import ConnectionPoint from '../ConnectionPoint';
+import { elementTypes, satisfiabilityTypes } from '../../constants';
+import translations from '../../translations.intl';
+import { nodeShape, selectedElementShape } from '../../propTypes';
+
+const styles = {
+  andGate: {
+    border: '1px solid black',
+    cursor: 'pointer',
+    position: 'relative',
+  },
+  andGateInput: {
+    border: '1px solid black',
+    backgroundColor: 'white',
+    height: '18px',
+    width: '8px',
+  },
+  orGate: {
+    cursor: 'pointer',
+    border: '1px solid black',
+    position: 'relative',
+  },
+  orGateInput: {
+    backgroundColor: 'white',
+    height: '18px',
+    width: '8px',
+  },
+  selectedGate: {
+    boxShadow: '0px 0px 2px 2px #3297fd',
+  },
+  summaryGate: {
+    backgroundColor: 'white',
+    border: '1px solid black',
+    cursor: 'pointer',
+    height: '20px',
+    padding: '0px 2px',
+    position: 'relative',
+  },
+  wrapper: {
+    alignItems: 'center',
+    display: 'flex',
+    flexDirection: 'row',
+    position: 'relative',
+  },
+};
+
+const Gate = (props) => {
+  const {
+    canModify,
+    dispatch,
+    gateInputSizeThreshold,
+    getGateConnectionPointId,
+    getGateId,
+    getGateInputId,
+    node,
+    selectedElement,
+  } = props;
+  const id = getGateId(node.id);
+  const zIndex = node.depth + 2;
+  const isSelected =
+    selectedElement.type === elementTypes.gate && selectedElement.id === id;
+
+  const onGateClick = (event) => {
+    if (canModify) {
+      event.stopPropagation();
+      dispatch(selectGate(id));
+    }
+  };
+
+  const isAndGate = () =>
+    node.satisfiabilityType === satisfiabilityTypes.allConditions;
+  const isSummaryGate = () => node.parents.length > gateInputSizeThreshold;
+
+  const getGateBackgroundColor = (isSatisfied) => {
+    if (canModify) {
+      return 'white';
+    }
+
+    return isSatisfied ? `${green[300]}` : `${red[300]}`;
```

Comment:
I don't think there's a need to interpolate - if I'm not wrong, `green[300]` is already a string (same for `red[300]`).

Reply:
Was used last time before I switched to MUI colours but forgot to remove the backticks. I've removed the string interpolation. Thanks for pointing it out!